### PR TITLE
M2P-256 Add support to enable PPC only on specific products

### PR DESCRIPTION
**Pull request by njfaries**
# Description
Adds a configuration field to allow merchants to enable PPC only on products with the `bolt_ppc` attribute.

Fixes: [M2P-256](https://boltpay.atlassian.net/browse/M2P-256)

#changelog Add support to enable PPC only on specific products

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Block/JsProductPage.php [hunk 1]
```diff
@@ -165,4 +165,18 @@ public function isSaveHintsInSections()
...
     }
+
+    public function isBoltProductPage()
+    {
+        if ($this->configHelper->getSelectProductPageCheckoutFlag())
+        {
+            $attributes = $this->getProduct()->getAttributes();
+            $codes = [];
+            foreach ($attributes as $attribute) {
+                array_push($codes, $attribute->getName());
```
Comment (vitaliyreznikov) [inline Block/JsProductPage.php:176]: We don't need other attributes so maybe we can do early return once we find 'bolt_ppc'

Comment (njfaries) [reply Block/JsProductPage.php]: The `getData` approach we discussed offline didn't work, but I've implemented a quicker check now

**Comment by vitaliyreznikov** [comment]
Could you also rebase from master and add new setting to new code @meganschnedar-bolt introduces in https://github.com/BoltApp/bolt-magento2/pull/994 ?

File: Block/JsProductPage.php [hunk 2]
```diff
@@ -165,4 +165,20 @@ public function isSaveHintsInSections()
     {
         return $this->featureSwitches->isSaveHintsInSections();
     }
+
+    public function isBoltProductPage()
+    {
+        if ($this->configHelper->getSelectProductPageCheckoutFlag())
```
[the remark this answers is not in the document]
Comment (njfaries) [reply Block/JsProductPage.php:171]: I agree with that change, will implement with update for Megan's PR

**Review by vitaliyreznikov — APPROVED** [review]

**Pull request merged**